packages/vite/src/node/plugins/vendorAnalysis.ts

```diff
@@ -0,0 +1,244 @@
+import type { ResolvedConfig } from '../config'
+import type { Plugin } from '../plugin'
+import type { ModuleInfo } from 'rollup'
+import type { ImportSpecifier } from 'es-module-lexer'
+import { init, parse as parseImports } from 'es-module-lexer'
+import { isCSSRequest } from './css'
+import { createDebugger, timeFrom } from '../utils'
+import { performance } from 'perf_hooks'
+
+const isDebug = !!process.env.DEBUG
+const debug = createDebugger('vite:vendor-analysis')
+
+/**
+ * Build only. Analyze which modules are used by entries and
+ * generate manualChunks config to split those files to vendor.
```

Comment:
this could use an explanation as to what the algorithm is and why this is being used instead of Rollup's default behavior

Reply:
added a simple explanation, Is it okay?